src/main/java/org/jitsi/jicofo/FocusBundleActivator.java

```diff
@@ -169,10 +211,20 @@ public static EventAdmin getEventAdmin()
     }
 
     /**
-     * Returns shared thread pool service.
+     * Returns a {@link ScheduledExecutorService} shared by all components
+     * through OSGi.
+     */
+    public static ScheduledExecutorService getSharedScheduledThreadPool()
+    {
+        return ServiceUtils2.getService(bundleContext, ScheduledExecutorService.class);
+    }
+
+    /**
+     * Returns a cached {@link ExecutorService} shared by all components through
+     * OSGi.
      */
-    public static ScheduledExecutorService getSharedThreadPool()
+    public static ExecutorService getSharedThreadPool()
     {
-        return sharedThreadPool;
+        return ServiceUtils2.getService(bundleContext, ExecutorService.class);
```

Comment:
Registering a service with such a generic class seems like a bad idea. It seems like all usage is restricted to this class (`FocusBundleActivator`), so why not bypass OSGi and return a reference to `cachedPool` directly?

The same applies `getSharedScheduledThreadPool()`

Reply:
Actually, is there any code which uses `getSharedThreadPool`?

Reply:
https://github.com/jitsi/jicofo/blob/0b0905e34fb5114b04b5263817e11e3de7ed350c/src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java#L926
and
https://github.com/jitsi/jicofo/blob/0b0905e34fb5114b04b5263817e11e3de7ed350c/src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java#L3094
So it's both places that create channel allocators, which  is what the commit msg says: moves channel allocators from bounded size pool to unbounded, cached pool.

Reply:
> so why not bypass OSGi and return a reference to cachedPool directly?

It's just that there is code that doesn't deal with the `FocusBundleActivator` directly, but retrieves it through OSGi.

Reply:
I have removed registration of `ExecutorService` as the only usages were going through the bundle activator anyway.